src/ui/link/index.tsx

```diff
@@ -83,13 +87,24 @@ const Link = forwardRef<HTMLButtonElement | HTMLAnchorElement, LinkProps>(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [route, linkDestination, routerActions]);
 
+    const schedulePrefetch = useCallback(() => {
```

Comment:
Nit: Wonder if we could optimise this a bit by having a smarter `prefetchRef` object instead of multiple `useCallback`:
```
const prefetchRef = useRef({
  timer: null,
  schedule() { ... },
  cancel() { ... },
})
```